server/models/procedures/stock.sql

```diff
@@ -274,4 +274,188 @@ BEGIN
   VALUES (HUID(UUID()), documentUuid, depotUuid, lotUuid, fluxId, CURRENT_DATE(), stockLotQuantity, inventoryUnitCost, 0, userId, periodId);
 END $$
 
+-- the main procedure that loops through stock_movement , retrieve and caculate cmm data
+/*
+CALL computeStockQuantity(startDate, inventoryUuid, depotUuid);
+
+DESCRIPTION
+Computes the quantity in stock from the startDate until now for a given inventoryUuid.
+
+Originally written by @jeremielodi
+*/
+DROP PROCEDURE IF EXISTS `computeStockQuantity`$$
+CREATE PROCEDURE `computeStockQuantity` (
+  IN _start_date DATE,
+  IN _inventory_uuid BINARY(16),
+  IN _depot_filter_uuid BINARY(16)
+) BEGIN
+  DECLARE done BOOLEAN;
+  DECLARE _depot_uuid, _row_uuid BINARY(16);
+  DECLARE _inventory_name TEXT;
+  DECLARE _end_date DATE;
+  DECLARE _qtt, _in_qtt, _out_qtt DECIMAL(19, 4);
+
+  -- DROP TEMPORARY TABLE IF EXISTS `stock_movement_status`;
```

Comment:
Don't you want to drop this table in the main routine above?  Or is "temporary table" enough to avoid reusing the table between queries?

Reply:
It's should be a permanent table, I used a temporary table just for the demostration

Reply:
@jniles we don't need to store the ```inventory name``` as we already have it ```uuid```

Reply:
Thanks for the explanations.  That helped me understand what you were doing here.